pkg/vm/engine/disttae/table.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package disttae
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/matrixorigin/matrixone/pkg/container/batch"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/vm/engine"
+)
+
+var _ engine.Relation = new(table)
+
+func (tbl *table) Rows() int64 {
+	return 0
+}
+
+func (tbl *table) Size(name string) int64 {
+	return 0
+}
+
+func (tbl *table) Ranges(ctx context.Context) ([][]byte, error) {
```

Comment:
What is Ranges?

Reply:
> What is Ranges?


For tae it is the block list, for memengine it is the range of primary keys, like [0 - 10), [10, 20), [20, 30), [30, max].
This interface is mainly used to slice and dice queries into different pipelines. @fengttt